pkg/apis/litmuschaos/v1alpha1/chaosgraph_types.go

```diff
@@ -0,0 +1,68 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ChaosGraphSpec defines the desired state of ChaosGraph
+// +k8s:openapi-gen=true
+// ChaosGraph defines the low-level chaos-params and the executor infrastructure
+type ChaosGraphSpec struct {
+        // Defines the chaos executor framework/infrastructure to run the experiment
+        Executor     string              `json:"executor"`
+        // Defines the chaos parameters and execution artifacts 
+        Definition   ChaosRunDefinition  `json:"definition"`
+}
+
+// ChaosRunDefinition defines the information fed to the executor framework
+// to enable successful chaos injection
+type ChaosRunDefinition struct {
+        // Default labels of the executor pod
+        // +optional
+        Labels       map[string]string   `json:"labels"`
```

Comment:
Should we have `Image` as one of the definition options?

Reply:
That could definitely be one way... Another would be to delegate the image to the executor infrastructure.. considering that the chaosgraphs are also pulled by charts..